pkg/policy/validate.go

```diff
@@ -425,6 +422,39 @@ func Validate(policy kyvernov1.PolicyInterface, client dclient.Interface, mock b
 	return warnings, nil
 }
 
+func UpdateSourceResource(client dclient.Interface, kind, namespace string, policy kyvernov1.PolicyInterface, obj *unstructured.Unstructured) error {
+	updateSource := true
+	label := obj.GetLabels()
+
+	if len(label) == 0 {
+		label = make(map[string]string)
+		label["generate.kyverno.io/clone-policy-name"] = policy.GetName()
+	} else {
+		if label["generate.kyverno.io/clone-policy-name"] != "" {
+			policyNames := label["generate.kyverno.io/clone-policy-name"]
+			if !strings.Contains(policyNames, policy.GetName()) {
+				policyNames = policyNames + "," + policy.GetName()
+				label["generate.kyverno.io/clone-policy-name"] = policyNames
+			} else {
+				updateSource = false
+			}
+		} else {
+			label["generate.kyverno.io/clone-policy-name"] = policy.GetName()
+		}
+	}
+
+	if updateSource {
+		logging.V(4).Info("updating existing clone source labels")
+		obj.SetLabels(label)
+		_, err := client.UpdateResource(obj.GetAPIVersion(), kind, namespace, obj, false)
+		if err != nil {
+			logging.Error(err, "failed to update source", "kind", obj.GetKind(), "name", obj.GetName(), "namespace", obj.GetNamespace())
+		}
+		logging.V(4).Info("updated source", "kind", obj.GetKind(), "name", obj.GetName(), "namespace", obj.GetNamespace())
```

Comment:
This message is printed regardless of the error value, please change it.

Reply:
Yes, added the else there, did not pushed the latest changes yet.